### PR TITLE
Run forum tests in CI

### PR DESCRIPTION
**Pull request by achamayou**
Virtual mode and CFT only right now, adding more flag passing to enable the rest.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -865,4 +865,18 @@ if(BUILD_TESTS)
       10000
       --use-websockets
   )
+
+  if(BUILD_END_TO_END_TESTS)
+    if("virtual" IN_LIST COMPILE_TARGETS)
+      add_test(NAME forum_app
+               COMMAND ${CMAKE_CURRENT_LIST_DIR}/samples/apps/forum/ci.sh
```
Comment (eddyashton) [inline CMakeLists.txt:872]: Missing file from this PR.

Comment (achamayou) [reply CMakeLists.txt:872]: Yep, sorry. It's 2020, and not only does my car not drive itself, but my RCS can't even tell what I mean. I want my AI money back.

File: CMakeLists.txt [hunk 2]
```diff
@@ -822,6 +822,17 @@ if(BUILD_TESTS)
     ADDITIONAL_ARGS -p liblogging --duration 45m
   )
 
+  add_e2e_sandbox_test(
```
Comment (eddyashton) [inline CMakeLists.txt:825]: If you add an entry for this to `.cmake-format.py` it will be forwarded like our other CMake functions.

**Review by eddyashton — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (achamayou) [reply CMakeLists.txt:825]: Got it, it's forwarded now :)

**Review by eddyashton — APPROVED** [review]

**Pull request merged**